nucleus/track.py

```diff
@@ -0,0 +1,166 @@
+        """Serializes track object to schematized JSON string."""
+        return json.dumps(self.to_payload(), allow_nan=False)
+
+    def update(
+        self,
+        scene_reference_id: Optional[str] = None,
+        metadata: Optional[dict] = None,
+        overwrite_metadata: bool = False,
+    ) -> None:
+        """
+        Updates the Track's scene_reference_id or metadata.
+
+        Parameters:
+            scene_reference_id (Optional[str]): The reference ID of the scene this track links to.
+            metadata (Optional[dict]): An arbitrary dictionary of additional data about this track that can be stored
+                and retrieved.
+            overwrite_metadata (Optional[bool]): If metadata is provided and overwrite_metadata = True, then the track's
+                entire metadata object will be overwritten. Otherwise, only the keys in metadata will be overwritten.
+        """
+
+        assert (
+            scene_reference_id is not None and metadata is not None
+        ), "Must provide scene_reference_id or metadata"
+
+        self._client.make_request(
+            payload={
+                REFERENCE_ID_KEY: self.reference_id,
+                SCENE_REFERENCE_ID_KEY: scene_reference_id,
+                METADATA_KEY: metadata,
+                OVERWRITE_KEY: overwrite_metadata,
+            },
+            route=f"dataset/{self.dataset_id}/track/update",
+            requests_command=requests.post,
+        )
+        self.scene_reference_id = (
+            scene_reference_id
+            if scene_reference_id
+            else self.scene_reference_id
+        )
+        self.metadata = (
+            metadata
+            if overwrite_metadata
+            else (
+                {**self.metadata, **metadata}
+                if self.metadata is not None and metadata is not None
+                else metadata
+            )
+        )
+
+    def link(
+        self,
+        annotations: Optional[List[Annotation]] = None,
+        predictions: Optional[List[Prediction]] = None,
+    ) -> None:
+        """
+        Links annotations and predictions to the track. This allows the linked annotations and predictions
+        to be identified as instances of the track.
+
+        Parameters:
+            annotations (Optional[List[Annotation]]): The annotations to link.
+            predictions (Optional[List[Prediction]]): The predictions to link.
+        """
+
+        self._client.make_request(
+            payload={
+                REFERENCE_ID_KEY: self.reference_id,
+                ANNOTATIONS_KEY: [ann.reference_id for ann in annotations]
+                if annotations
+                else [],
+                PREDICTIONS_KEY: [pred.reference_id for pred in predictions]
+                if predictions
+                else [],
+            },
+            route=f"dataset/{self.dataset_id}/track/link",
+            requests_command=requests.post,
+        )
+
+    def unlink(
```

Comment:
To reiterate my prior thought: I think we should try our best to make tracks immutable. A track probably has a level as well, it doesn't ever have annotation and predictions in the same track.